mos-platform/atari8-mega/link.ld

```diff
@@ -0,0 +1,209 @@
+  bank22        : ORIGIN = __bank22_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank23        : ORIGIN = __bank23_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank24        : ORIGIN = __bank24_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank25        : ORIGIN = __bank25_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank26        : ORIGIN = __bank26_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank27        : ORIGIN = __bank27_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank28        : ORIGIN = __bank28_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank29        : ORIGIN = __bank29_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank30        : ORIGIN = __bank30_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+  bank31        : ORIGIN = __bank31_lma, LENGTH = __cart_rom_size >= 512 ? 0x4000 : 0
+}
+
+REGION_ALIAS("c_writeable", ram)
+REGION_ALIAS("c_readonly", bank0)
+
+SECTIONS {
+  INCLUDE c.ld
+
+  .bank0     : { *(.bank0     .bank0.*)     } >bank0
+  .bank1     : { *(.bank1     .bank1.*)     } >bank1
+  .bank2     : { *(.bank2     .bank2.*)     } >bank2
+  .bank3     : { *(.bank3     .bank3.*)     } >bank3
+  .bank4     : { *(.bank4     .bank4.*)     } >bank4
+  .bank5     : { *(.bank5     .bank5.*)     } >bank5
+  .bank6     : { *(.bank6     .bank6.*)     } >bank6
+  .bank7     : { *(.bank7     .bank7.*)     } >bank7
+  .bank8     : { *(.bank8     .bank8.*)     } >bank8
+  .bank9     : { *(.bank9     .bank9.*)     } >bank9
+  .bank10    : { *(.bank10    .bank10.*)    } >bank10
+  .bank11    : { *(.bank11    .bank11.*)    } >bank11
+  .bank12    : { *(.bank12    .bank12.*)    } >bank12
+  .bank13    : { *(.bank13    .bank13.*)    } >bank13
+  .bank14    : { *(.bank14    .bank14.*)    } >bank14
+  .bank15    : { *(.bank15    .bank15.*)    } >bank15
+  .bank16    : { *(.bank16    .bank16.*)    } >bank16
+  .bank17    : { *(.bank17    .bank17.*)    } >bank17
+  .bank18    : { *(.bank18    .bank18.*)    } >bank18
+  .bank19    : { *(.bank19    .bank19.*)    } >bank19
+  .bank20    : { *(.bank20    .bank20.*)    } >bank20
+  .bank21    : { *(.bank21    .bank21.*)    } >bank21
+  .bank22    : { *(.bank22    .bank22.*)    } >bank22
+  .bank23    : { *(.bank23    .bank23.*)    } >bank23
+  .bank24    : { *(.bank24    .bank24.*)    } >bank24
+  .bank25    : { *(.bank25    .bank25.*)    } >bank25
+  .bank26    : { *(.bank26    .bank26.*)    } >bank26
+  .bank27    : { *(.bank27    .bank27.*)    } >bank27
+  .bank28    : { *(.bank28    .bank28.*)    } >bank28
+  .bank29    : { *(.bank29    .bank29.*)    } >bank29
+  .bank30    : { *(.bank30    .bank30.*)    } >bank30
+  .bank31    : { *(.bank31    .bank31.*)    } >bank31
+
+  /* This isn't pretty.  SIC! cartridges start with bank0 selected but
+    the lower 8 KiB unmapped.  The entire 16 KiB is needed, use
+    _cart_init to enable the lower-bank and optionally adjust
+    RAMTOP. */
+  .vector __bank0_lma + 0x3ffa - 14 : {
```

Comment:
Can we put this in an assembly file? You might be able to get some inspiration from NES targets, I think the MMC1 might have a similar mechanism.

Reply:
I was aware of that, but it ended up making things more complicated that I liked.

Do you consider this a blocker for now?

Reply:
@asiekierka PTAL